lib/handlebars/compiler/compiler.js

```diff
@@ -323,18 +323,28 @@ Compiler.prototype = {
 
   Hash: function(hash) {
     let pairs = hash.pairs,
-        i = 0,
+        i = 0, splat = null,
         l = pairs.length;
 
     this.opcode('pushHash');
-
     for (; i < l; i++) {
+
+      if (pairs[i].type === 'Splat') {
+        if (splat !== null) {
+          throw new Exception('Multiple splats are not supported yet');
```

Comment:
@kpdecker My first thought was to convert all `HashPairs` into `Splats` so that it could be done as a `merge` since duplication isn't banned with right-most precedence:
`{{foo bar="bar" bar="whoops" **bat **other}}` => `merge({bar: "bar"}, {bar: "whoops"}, {bat: "bat"}, {other: ...});`

That would make the way we think about each piece consistent, but what I'm discovering is that I don't know enough about the underlying Handlebars primitives to evaluate this approach.

Does that seem reasonable?